custom_components/tahoma/sensor.py

```diff
@@ -75,7 +78,18 @@ def unit_of_measurement(self):
         states = self.tahoma_device.active_states
 
         if CORE_TEMPERATURE_STATE in states:
-            # TODO Retrieve core:MeasuredValueType to understand if it is Celsius or Kelvin
+            measured_value = self.tahoma_device.attributes.get(CORE_MEASURED_VALUE_TYPE)
+
+            if measured_value is not None:
+                if "Celsius" in measured_value:
```

Comment:
So measured_value is a string that contains something else than Celsius?

Reply:
yes, in my case it `core:TemperatureInCelcius`

Reply:
Is it Celcius or Celsius? :D (just to be sure, I make this mistake so often that I wouldn't be surprised if Somfy developers did the same)

Reply:
damn you're write: 
![image](https://user-images.githubusercontent.com/5701372/86611116-7e283380-bfae-11ea-8cba-ac0becfd1859.png)